Microsoft.NET.Build.Containers/Registry.cs

```diff
@@ -217,12 +217,24 @@ public async Task Push(Image x, string name, string? tag, string baseName, Actio
         manifestUploadContent.Headers.ContentType = new MediaTypeHeaderValue(DockerManifestV2);
         var putResponse = await client.PutAsync(new Uri(BaseUri, $"/v2/{name}/manifests/{x.GetDigest(x.manifest)}"), manifestUploadContent);
         string putresponsestr = await putResponse.Content.ReadAsStringAsync();
-        putResponse.EnsureSuccessStatusCode();
+
+        if (!putResponse.IsSuccessStatusCode)
+        {
+            string jsonResponse = await putResponse.Content.ReadAsStringAsync();
+            throw new ContainerHttpException("Registry push failed.", putResponse.RequestMessage?.RequestUri?.ToString(), jsonResponse);
+        }
```

Comment:
This is useful - I think it would make sense to extract out to an extension on `HttpResponseMessage` so that would could easily treat all Registry HTTP calls in this manner.

Reply:
Not sure I'm on the same page, I read this as making ContainerHttpExtension inherit from `HttpResponseMessage`, is that what you meant?